adal/src/main/java/com/microsoft/aad/adal/Discovery.java

```diff
@@ -104,16 +104,17 @@ public boolean isValidAuthority(URL authorizationEndpoint) {
                         ADALError.DEVELOPER_AUTHORITY_CAN_NOT_BE_VALIDED,
                         new AuthenticationException(ADALError.DISCOVERY_NOT_SUPPORTED));
                 return false;
-            } else if (sValidHosts.contains(authorizationEndpoint.getHost().toLowerCase(Locale.US))) {
```

Comment:
IMO, this is not the way to avoid early returns. If you really want to avoid early return, do in this way, 
final boolean isValidAuthority;
if (...) {
    isValidAuthority = ...;
} else if () {
    isValidAuthority = ...;
} else {
    isValidAuthority = ...;
}

return isValidAuthority;

---
In reply to: [66152820](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/652#discussion_r66152820) [](ancestors = 66152820,66004394)